integration/spark/src/main/spark3/java/io/openlineage/spark3/agent/lifecycle/plan/V2CatalogParser.java

```diff
@@ -0,0 +1,25 @@
+package io.openlineage.spark3.agent.lifecycle.plan;
+
+import io.openlineage.spark.agent.util.PlanUtils;
+import lombok.SneakyThrows;
+import org.apache.commons.lang3.reflect.FieldUtils;
+import org.apache.hadoop.fs.Path;
+import org.apache.spark.sql.connector.catalog.CatalogPlugin;
+import org.apache.spark.sql.execution.datasources.jdbc.JDBCOptions;
+import org.apache.spark.sql.execution.datasources.v2.jdbc.JDBCTableCatalog;
+
+public class V2CatalogParser {
+  public static Path getCatalogLocation(CatalogPlugin catalog) {
```

Comment:
I think the `PlanUtils.getDataset(CatalogTable)` route that the other visitors use will offer some consistency over how we treat the datasources from a catalog. E.g., I don't know that "catalog location" makes sense for a lot of catalogs- e.g., in memory ones. And in Hive, the table location itself is used to determine the dataset, not the catalog's location.

Reply:
Addressed in https://github.com/OpenLineage/OpenLineage/pull/416 and https://github.com/OpenLineage/OpenLineage/pull/419